test/extended/util/cloud/cloud.go

```diff
@@ -0,0 +1,75 @@
+package cloud
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+)
+
+// LoadConfig uses the cluster to setup the cloud provider config.
+func LoadConfig() (string, *e2e.CloudConfig, error) {
+	coreClient, err := e2e.LoadClientset()
+	if err != nil {
+		return "", nil, err
+	}
+	clientConfig, err := e2e.LoadConfig()
+	if err != nil {
+		return "", nil, err
+	}
+	client := configclient.NewForConfigOrDie(clientConfig)
+
+	infra, err := client.ConfigV1().Infrastructures().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return "", nil, err
+	}
+	p := infra.Status.PlatformStatus
+	if p == nil {
+		return "", nil, fmt.Errorf("status.platformStatus must be set")
+	}
+	if p.Type == configv1.NonePlatformType {
+		return "", nil, nil
+	}
+
+	var provider string
+	var projectID string
+	var region string
+	switch {
+	case p.AWS != nil:
+		provider = "aws"
+		region = p.AWS.Region
+	case p.GCP != nil:
+		provider = "gcp"
+		projectID = p.GCP.ProjectID
+		region = p.GCP.Region
+	case p.Azure != nil:
+		provider = "azure"
```

Comment:
why stop here? why not setup the platform correctly when asked.

This assumes that somebody else will handle the azure stuff, if it's due to the tmp file handling it would be more appropriate to get that as option to this functions...

because when we need overrides like how to set service-endpoint-overrides for aws client when testing the config file could be handy..

Reply:
The caller covers that today, I can move it in here, but this is a utility method for the default.